packages/core/src/render3/instructions/shared.ts

```diff
@@ -505,7 +505,7 @@ export function refreshView<T>(
 export function renderComponentOrTemplate<T>(
     tView: TView, lView: LView, templateFn: ComponentTemplate<{}>|null, context: T) {
   const rendererFactory = lView[RENDERER_FACTORY];
-  const normalExecutionPath = !isInCheckNoChangesMode();
+  const normalExecutionPath = !ngDevMode || !isInCheckNoChangesMode();
```

Comment:
nit: may be rename the flag and invert the condition, so it's more obvious what this is (+ add a comment, similar to the code above):
```suggestion
  // Check no changes mode is a dev only mode used to verify that bindings have not changed
  // since they were assigned. We do not want to invoke renderer factory functions in that mode
  // to avoid any possible side-effects.
  const devModeExecutionPath = !!ngDevMode && isInCheckNoChangesMode();
```

Reply:
I opted for `checkNoChangesMode` as including `devMode` in the name is confusing to me, as the "primary" execution also runs in dev mode.

Reply:
Makes sense, thanks Joost!